app/utils/sql/SqlInterpolation.scala

```diff
@@ -102,16 +94,12 @@ object SqlToken {
     SqlToken(sql = outputSql.toString, values = outputValues.toList)
   }
 
-  def tuple(values: Seq[Any]): SqlToken = {
-    val sqlValues = values.map(SqlValue.makeSqlValue)
-    SqlToken(sql = s"(${sqlValues.map(_.placeholder).mkString(", ")})", values = sqlValues.toList)
-  }
+  def tuple(values: List[SqlValue]): SqlToken =
+    SqlToken(sql = s"(${values.map(_.placeholder).mkString(", ")})", values = values)
```

Comment:
```suggestion
  def tupleFromList(values: List[SqlValue]): SqlToken =
    SqlToken(sql = s"(${values.map(_.placeholder).mkString(", ")})", values = values)
  def tuple(values: SqlValue*): SqlToken =
    SqlToken(sql = s"(${values.map(_.placeholder).mkString(", ")})", values = values.toList)  
```

Reply:
How important do you think this is? My concern is that one might accidentally use tuple instead of tupleFromList. It will compile fine (as List[SqlValue] has an implicit conversion to SqlValue), but will not perform as expected. I currently see no usages where tuple is called in this way.

The nested one seems more relevant, but there this trick doesn’t work :grimacing:

Reply:
I think that would improve the readability. I used that in #6732 for INSERTs, e.g. https://github.com/scalableminds/webknossos/pull/6732/files#diff-a2ec134373baf5603901ac361985d256c48f4f0e17025a96c4e5d6f0fec98c5fR945-R955, and for inline tuples, e.g. https://github.com/scalableminds/webknossos/pull/6732/files#diff-a2ec134373baf5603901ac361985d256c48f4f0e17025a96c4e5d6f0fec98c5fR464.

We can debate naming, of course.

`tupleLists` could be replaced by `SqlToken.join(list.map(i => SqlToken.tuple(i._1, i._2)), ",")`. Could `Iterable[SqlToken]` be extended to have a `toTupleList` function (with implicit magic)?

Reply:
Fair enough. I also renamed tuple to tupleFromValues to explicitly distinguish it from tupleFromLists.